Extensions/AdMob/admobtools.ts

```diff
@@ -93,288 +172,415 @@ namespace gdjs {
         | 'LEADERBOARD'
         | 'SMART_BANNER'
     ) => {
-      const adSizeTypes = {
-        BANNER: AdSizeType.BANNER,
-        LARGE_BANNER: AdSizeType.LARGE_BANNER,
-        MEDIUM_RECTANGLE: AdSizeType.MEDIUM_RECTANGLE,
-        FULL_BANNER: AdSizeType.FULL_BANNER,
-        LEADERBOARD: AdSizeType.LEADERBOARD,
-        SMART_BANNER: AdSizeType.SMART_BANNER,
-      };
-
       bannerRequestedAdSizeType =
         adSizeTypes[bannerAdSizeType] || AdSizeType.SMART_BANNER;
     };
+
     /**
-     * Display the banner that was set up with `loadBanner` (and `setBannerAdSizeType`).
+     * Set up a banner that can then be displayed by calling `showBanner`.
+     * If a banner is already set up, it will be hidden and replaced by the new one.
      */
-    export const showBanner = () => {
-      if (typeof admob === 'undefined') {
-        return;
+    export const setupBanner = async (androidAdUnitId, iosAdUnitId, atTop) => {
+      if (!checkIfAdMobIsAvailable()) return;
+      const adUnitId = getAdUnitId(androidAdUnitId, iosAdUnitId, 'banner');
+      if (!adUnitId) return;
+
+      if (banner && bannerShowing) {
+        logger.info('Banner already visible, hiding it to display new one.');
+        await hideBanner();
       }
 
-      bannerLoading = true;
-      bannerShowing = false;
+      bannerConfigured = false;
+      bannerLoaded = false;
+
+      banner = new admob.BannerAd({
+        adUnitId,
+        position: atTop ? 'top' : 'bottom',
+        size: bannerRequestedAdSizeType,
+      });
+
+      banner.on('load', () => {
+        bannerShowing = true;
+        bannerLoaded = true;
+      });
+      banner.on('load_fail', () => {
```

Comment:
🚨 This should be 
```suggestion
      banner.on('loadfail', () => {
```

according to https://admob-plus.github.io/docs/cordova/ads/banner#loadfail and https://github.com/admob-plus/admob-plus/search?q=loadfail

Reply:
oh yes forgot to update this one, nice catch